src/index.ts

```diff
@@ -2013,7 +2015,10 @@ export class GossipSub extends EventEmitter<GossipsubEvents> implements PubSub<G
     const msgId = await this.msgIdFn(msg)
     const msgIdStr = this.msgIdToStrFn(msgId)
 
-    if (this.seenCache.has(msgIdStr)) {
+    // Current publish opt takes precedence global opts, while preserving false value
+    const allowPublishDuplicates = opts?.allowPublishDuplicates ?? this.opts.allowPublishDuplicates
+
+    if (this.seenCache.has(msgIdStr) && !allowPublishDuplicates) {
```

Comment:
it's no use to publish again if there is a duplicate, just return here. In that case, I think we might want to go with a better name, `ignorePublishDuplicate` or any suggestions @wemeetagain ?

also let's add a gauge metric to track this with topic as a label

Reply:
Thanks for clarifying @tuyennhv  I think I had misunderstood, I would say `ignoreDuplicateMessages` would be a more appropriate label. I will also integrate the metrics